ga4gh/datamodel/reads.py

```diff
@@ -23,14 +85,21 @@ def __init__(self, id_, dataDir):
         self._id = id_
         self._dataDir = dataDir
         self._readGroups = []
-        for fileType in ["sam", "bam"]:
-            pattern = "*.{}".format(fileType)
-            for path in glob.glob(os.path.join(self._dataDir, pattern)):
-                filename = os.path.split(path)[1]
-                localId = filename.split(".")[0]
-                readGroupId = "{}:{}".format(self._id, localId)
-                readGroup = ReadGroup(readGroupId, path)
-                self._readGroups.append(readGroup)
+        # we only support BAM files right now;
+        # SAM files (which can't have index files) would be too slow
+        pattern = "*.bam"
+        for path in glob.glob(os.path.join(self._dataDir, pattern)):
+            filename = os.path.split(path)[1]
```

Comment:
I personally prefer `os.path.basename(path)` unless the other part of the split will be used as well.